app/models/app_remote_kill_migration.rb

```diff
@@ -0,0 +1,70 @@
+class AppRemoteKillMigration
+  include ActiveModel::Validations
+
+  attr_reader :app,
+    :split,
+    :reason,
+    :first_bad_version_before_type_cast
+
+  validate :split_must_exist
+  validates :reason, presence: true
+
+  delegate :override_to,
+    :first_bad_version,
+    :fixed_version,
+    to: :app_remote_kill
+
+  def initialize(params)
+    @app = params[:app] || raise("Must provide app")
+    @split = params[:split]
+    @reason = params[:reason]
+    @first_bad_version_before_type_cast = params[:first_bad_version]
+    app_remote_kill.assign_attributes(
+      override_to: params[:override_to],
+      first_bad_version: params[:first_bad_version],
+      fixed_version: params[:fixed_version]
+    )
+  end
+
+  def valid?
+    if destroy?
+      super
+    else
+      app_remote_kill.valid?
+    end
+  end
+
+  def errors
+    if destroy?
+      super
+    else
+      app_remote_kill.errors
+    end
+  end
+
+  def save
+    if destroy?
+      valid? && app_remote_kill.destroy.destroyed?
+    else
+      app_remote_kill.save
+    end
+  end
+
+  def destroy?
```

Comment:
Does this need to be public? (Same for feature completions)

Reply:
it doesn't need to be, but my general take is that if I'm not embarrassed by it being depended on by a collaborator or test, I will tend to go public. Turns out I didn't test it directly, but I wouldn't have felt bad doing that.